be/src/vec/aggregate_functions/aggregate_function_sort.h

```diff
@@ -90,7 +90,11 @@ struct AggregateFunctionSortData {
         PBlock pblock;
         pblock.ParseFromString(data);
         auto st = block.deserialize(pblock);
-        CHECK(st.ok());
+        // If memory allocate failed during deserialize, st is not ok, throw exception here to
+        // stop the query.
+        if (!st.ok()) {
+            throw doris::Exception(ErrorCode::INTERNAL_ERROR, st.to_string());
```

Comment:
we have more proper ErrorCode MemoryLimitExceeded

Reply:
throw doris::Exception(st.code(), st.to_string());